include/mgp_mock.py

```diff
@@ -0,0 +1,1916 @@
+
+        Raises:
+            InvalidContextError: If the graph is not in a valid context.
+            ImmutableObjectError: If the graph is immutable.
+            DeletedObjectError: If `from_vertex` or `to_vertex` have been deleted.
+
+        Examples:
+            ```edge = graph.create_edge(from_vertex, vertex, edge_type)```
+        """
+
+        if not self.is_valid():
+            raise InvalidContextError()
+
+        if self._graph.is_immutable():
+            raise ImmutableObjectError("Cannot modify immutable object.")
+
+        new_edge = self._graph.create_edge(from_vertex._vertex, to_vertex._vertex, edge_type.name)
+        return Edge(new_edge)
+
+    def delete_edge(self, edge: Edge) -> None:
+        """
+        Delete the given edge.
+
+        Args:
+            edge: The `Edge` to be deleted.
+
+        Raises:
+            InvalidContextError: If the graph is not in a valid context.
+            ImmutableObjectError: If the graph is immutable.
+
+        Examples:
+            ```graph.delete_edge(edge)```
+        """
+        if not self.is_valid():
+            raise InvalidContextError()
+
+        if self._graph.is_immutable():
+            raise ImmutableObjectError("Cannot modify immutable object.")
+
+        self._graph.delete_edge(edge.from_vertex.id, edge.to_vertex.id, edge.id)
+
+
+class AbortError(Exception):
+    """Signals that the procedure was asked to abort its execution."""
+
+    pass
+
+
+class ProcCtx:
+    """The context of the procedure being executed.
+
+    Access to a `ProcCtx` is only valid during a single execution of a query procedure.
+    You should not globally store a `ProcCtx` instance.
+    """
+
+    __slots__ = ("_graph",)
+
+    def __init__(self, graph):
+        if not isinstance(graph, _mgp_mock.Graph):
+            raise TypeError(f"Expected '_mgp_mock.Graph', got '{type(graph)}'")
+
+        self._graph = Graph(graph)
+
+    def is_valid(self) -> bool:
+        """
+        Check if the context is valid, i.e. if the contained structures may be used.
+
+        Returns:
+            A `bool` value that represents whether the context is valid.
+
+        Examples:
+            ```context.is_valid()```
+        """
+        return self._graph.is_valid()
+
+    @property
+    def graph(self) -> Graph:
+        """
+        Access the graph.
+
+        Returns:
+            A `Graph` object representing the graph.
+
+        Raises:
+            InvalidContextError: If the procedure context is not valid.
+
+        Examples:
+            ```context.graph```
+        """
+        if not self.is_valid():
+            raise InvalidContextError()
+
+        return self._graph
+
+    # TODO try to implement
+    # def must_abort(self) -> bool:
+    #     if not self.is_valid():
+    #         raise InvalidContextError()
+
+    #     return self._graph._graph.nx.must_abort()
+
+    # def check_must_abort(self):
+    #     if self.must_abort():
+    #         raise AbortError
```

Comment:
just comment that you are aware of this code

Reply:
Aware :raised_back_of_hand:

Reply:
@antepusic should we remove this?

Reply:
Yes, I removed this because must_abort is a DB-specific feature and thus not a part of the mock API